src/etc/inc/interfaces_fast.inc

```diff
@@ -0,0 +1,271 @@
+<?php
+/*
+ * interfaces_fast.inc
+ *
+ * part of pfSense (https://www.pfsense.org)
+ * Copyright (c) 2004-2016 Rubicon Communications, LLC (Netgate) (pfSense)
+ * Copyright (c) 2017 Peter Schofield (parts of this file)
+ * All rights reserved.
+ *
+ * originally based on m0n0wall (http://m0n0.ch/wall)
+ * Copyright (c) 2004 Manuel Kasper <mk@neon1.net>.
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ * This file contains rewrites of several functions, from both interfaces.inc
+ * and interfaces_assign.php. The general purpose of these rewrites is not
+ * necessarily to be faster in and of themselves, though they may be, but to
+ * replace functions called multiple times with a function that's called only
+ * once. This results in a significant speedup because there are far fewer
+ * function calls, fewer loops and suchlike. It does, however, increase memory
+ * usage somewhat, but it shouldn't be significant in the grand scheme of things.
+ *
+ *
+ * Functions in this file may not require/use parameters when called, however 
+ * in most cases they will accept parameters as per their original forms, for 
+ * consistency.
+ 
+ */
+require_once('interfaces.inc');
+/*
+* does_interface_exist_fast
+Returns an array of interrfaces which exist on the system.
+The $interface parameter is not used, but is accepted for
+consistency with the function it replaces.
+*/
+function does_interface_exist_fast($interface='', $flush = true) {
+	global $config;
+
+	$ints = get_interface_arr($flush);
+	return array_flip($ints);
+}
+
+/*
+ * convert_real_interface_to_friendly_interface_name_fast($interface): convert fxp0 -> wan, etc.
+ Returns an array of interfaces and friendly names.
+ Doesn't actually use any parameters, they're just accepted for consistency with the function it replaces
+ */
```

Comment:
I don't think this consistency is required, and it rather obfuscates the code. Could these arguments be removed wherever used?